### PR TITLE
Add new page Calibration type 10 details 

### PR DESCRIPTION
**Pull request by MSECode**
Update calibration type 10 doc
Add state machine for showing pipeline of hard-stop calibration when using AMO encoder at joint plus quadrature encoder at motor Add some line detailing the steps of the state machine

This PR aims at adding new details to the calibration type 10, which is the calibration of a joint to a defined hard-stop position, for joints provided with an AMO encoder at the joint and a quadrature encoder at the motor.
Basically a state machine diagram showing the calibration pipeline is shown and described so that now it is easier for somebody to know what is a joint that implements that calibration is doing.

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -277,6 +277,7 @@ nav:
         - iCub 3 Stator-Rotor auto-calibration: icub_robot_calibration/icub3-stator-rotor-autocalibration.md
       - Calibration Types:
         - Standard Calibration types: robot_calibration_types/standard_calibration_types.md
+        - AMO Encoder joint calibration type 10 details: robot_calibration_types/amo-encoder-joint-calibration-10-details.md
```
Comment (pattacini) [inline mkdocs.yml:280]: Adjust the name of the file using underscores.

File: docs/robot_calibration_types/standard_calibration_types.md [hunk 2]
```diff
@@ -140,6 +140,8 @@ This calibration is used for joints with relative encoders, which are calibrated
 
 Regarding iCub 3 and the AMO boards, the AMO magnetic target is subdivided into 64 or 32 sectors. The AMO sensor provides a 14-bit resolution per sector, for a total of 14+6=20 or 14+5=19 bits per revolution resolution, respectively. The `Gearbox_E2J` parameter must be configured with the number of sectors in the corresponding `mechanicals/part-mec.xml` file.
 
+Further details to how the workflow of the whole calibration type 10 is managed are described at [this page](./amo-encoder-joint-calibration-10-details.md)
```
Comment (pattacini) [inline docs/robot_calibration_types/standard_calibration_types.md:143]:
```suggestion
Further details of how the workflow of the whole calibration type 10 is managed are outlined on [this page](./amo-encoder-joint-calibration-10-details.md).
```

File: docs/robot_calibration_types/amo-encoder-joint-calibration-10-details.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+## Calibration workflow
+
+Complete pipeline for the `calibration type 10` using `AMO encoder` at joint and `quadrature encoder` at motor.
+<p align="center">
+    <img  src="../img/calibration-amo-abs-enc.png" width="700">
+</p>
+
+As you can see in the above reported state machine, after the calibration command is received by the EMS, it arms a timer of 30 seconds, then it waits for the motor encoder calibration ending and then starts the calibration 10 procedure.
+
+It basically consists in moving the slow shaft towards one of the hardware limit, which is the position set in the calibration parameter `calibrationZero` in the configuration files, with a value of PWM as the one defined in the parameter `calibration1` until the joint cannot move anymore inside a time windows and a space window, whose values are defined in the firmware.
```
Comment (pattacini) [inline docs/robot_calibration_types/amo-encoder-joint-calibration-10-details.md:17]:
```suggestion
The procedure basically consists in moving the slow shaft towards one of the hardware limits as defined in the position set within the calibration parameter `calibrationZero`. The value of the PWM used is in turn defined through the parameter `calibration1`. The joint moves until the limit is reached within a time window and within a predefined maximum length, whose values are coded in the firmware.
```

File: docs/robot_calibration_types/amo-encoder-joint-calibration-10-details.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+Finally, if the `hard-stop` limit position is found before the `CALIBRATION TIMEOUT` expires the zero of the `AMO encoder` is updated with the found offset and the whole calibration completes correctly, otherwise the joint is moved to the `NOT CONFIGURED` state, the PWM is set to `zero` and the motor moved to `IDLE`.
```
Comment (pattacini) [inline docs/robot_calibration_types/amo-encoder-joint-calibration-10-details.md:19]:
```suggestion
Finally, if the `hard-stop` limit position is found before the `CALIBRATION TIMEOUT` expires, the zero of the `AMO encoder` is updated with the found offset and the whole calibration completes correctly; otherwise, the joint is moved to the `NOT CONFIGURED` state, the PWM is set to `zero` and the motor switched to `IDLE`.
```

File: docs/robot_calibration_types/amo-encoder-joint-calibration-10-details.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+The fw checks if the joint has reached the hard position by checking that the encoder position does not change for 1 second, that means the relative increase of joint position is smaller than 6.5 deg (1200 iCubDegrees ) within 1 second. The encoder position taken into account for the check is already filtered from the spikes.
```
Comment (pattacini) [inline docs/robot_calibration_types/amo-encoder-joint-calibration-10-details.md:21]:
```suggestion
The FW verifies if the joint has reached the hard position by checking that the encoder position does not change for 1 second, meaning that the relative increase of joint position is smaller than 6.5 deg (1200 iCubDegrees) within 1 second. The encoder position considered for the check is processed to filter out the spikes.
```

**Review by pattacini — CHANGES_REQUESTED** [review]
Hi @MSECode 

Thanks for the new page!
Some inline comments to address.

File: docs/robot_calibration_types/amo-encoder-joint-calibration-10-details.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# Calibration type 10 details - AMO Encoder Joint Calibration
+
+This page shows how the calibration procedure used for `iCub 3/ergoCub` joints that mount an AMO encoder at the joint works.
+As already detailed in the [`calibration-10-procedure`](https://icub-tech-iit.github.io/documentation/robot_calibration_types/standard_calibration_types/#calibration-10) joints provided with the `AMO` encoder are currently calibrated so that it can be handled as an incremental encoder coupled with a quadrature encoder placed at the motor.
```
[the remark this answers is not in the document]
Comment (MSECode) [reply docs/robot_calibration_types/amo-encoder-joint-calibration-10-details.md]: Ok, I updated it.

File: docs/robot_calibration_types/amo-encoder-joint-calibration-10-details.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+This page aims at providing a detailed explanation to the steps performed by the controller while calibrating a joint-set as defined above.
+
+## Calibration workflow
+
+Complete pipeline for the `calibration type 10` using `AMO encoder` at joint and `quadrature encoder` at motor.
+<p align="center">
+    <img  src="../img/calibration-amo-abs-enc.png" width="700">
+</p>
+
+As you can see in the above reported state machine, after the calibration command is received by the EMS, it arms a timer of 30 seconds, then it waits for the motor encoder calibration ending and then starts the calibration 10 procedure.
```
[the remark this answers is not in the document]
Comment (MSECode) [reply docs/robot_calibration_types/amo-encoder-joint-calibration-10-details.md]: Updated and squashed commits

File: docs/robot_calibration_types/standard_calibration_types.md (hunk 2 above)
Comment (MSECode) [reply docs/robot_calibration_types/standard_calibration_types.md]: Updated and squashed commits

File: docs/robot_calibration_types/amo-encoder-joint-calibration-10-details.md (hunk 3 above)
Comment (MSECode) [reply docs/robot_calibration_types/amo-encoder-joint-calibration-10-details.md]: Updated and squashed commits

File: docs/robot_calibration_types/amo-encoder-joint-calibration-10-details.md (hunk 4 above)
Comment (MSECode) [reply docs/robot_calibration_types/amo-encoder-joint-calibration-10-details.md]: Updated and squashed commits

File: docs/robot_calibration_types/amo-encoder-joint-calibration-10-details.md (hunk 5 above)
Comment (MSECode) [reply docs/robot_calibration_types/amo-encoder-joint-calibration-10-details.md]: Updated and squashed commits

File: docs/robot_calibration_types/amo_encoder_joint_calibration_10_details.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
+# Calibration type 10 details - AMO Encoder Joint Calibration
+
+This page shows how the calibration procedure used for `iCub 3/ergoCub` joints that mount an AMO encoder at the joint works.
```
Comment (pattacini) [inline docs/robot_calibration_types/amo_encoder_joint_calibration_10_details.md:3]:
```suggestion
This section illustrates the details of the calibration procedure used for the `iCub 3/ergoCub` joints that mount an AMO encoder at the slow shaft.
```

File: docs/robot_calibration_types/amo_encoder_joint_calibration_10_details.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+As already detailed in the [`calibration-10-procedure`](./standard_calibration_types.md/#calibration-10) joints provided with the `AMO` encoder are currently calibrated so that it can be handled as an incremental encoder coupled with a quadrature encoder placed at the motor.
+
+## Calibration workflow
+
+Complete pipeline for the `calibration type 10` using `AMO encoder` at joint and `quadrature encoder` at motor.
+<p align="center">
+    <img  src="../img/calibration-amo-abs-enc.png" width="700">
+</p>
+
+As you can see in the above-reported state machine, after the calibration command is received by the EMS, a timer of 30 seconds gets triggered; afterward, we want untile the motor encoder calibration ends; finally, the calibration procedure 10 commences.
```
Comment (pattacini) [inline docs/robot_calibration_types/amo_encoder_joint_calibration_10_details.md:13]:
```suggestion
As you can see in the above-reported state machine, after the calibration command is received by the EMS, a timer of 30 seconds gets triggered; afterward, we wait until the motor encoder calibration ends; finally, the calibration procedure 10 commences.
```

**Comment by MSECode** [comment]
> amendments

Thanks
